saleor/checkout/utils.py

```diff
@@ -656,28 +661,36 @@ def change_shipping_address_in_checkout(checkout, address):
         checkout.save(update_fields=["shipping_address"])
 
 
-def get_checkout_context(checkout, discounts, currency=None, shipping_range=None):
+def get_checkout_context(
+    checkout: Checkout,
+    discounts: "DiscountsListType",
+    currency: Optional[str] = None,
+    shipping_range=None,
+) -> dict:
     """Retrieve the data shared between views in checkout process."""
-    manager = get_extensions_manager()
-    checkout_total = (
-        manager.calculate_checkout_total(checkout=checkout, discounts=discounts)
-        - checkout.get_total_gift_cards_balance()
-    )
+    cards_balance = checkout.get_total_gift_cards_balance()
+
+    # FIXME: We are having here parts of checkout total calculation logic
```

Comment:
Maybe we should put the substractions below to the `checkout_total` function? It could take gift card or gift card balance as an argument. Or create another calculation metod to substract it. It would be really nice to have this piece in a atomic and tested function.

Reply:
This method is used for storefront views context. So we will remove it soon.

To handle this logic in one place: https://github.com/mirumee/saleor/issues/5009